esti/hooks_test.go

```diff
@@ -42,6 +43,29 @@ func TestHooksSuccess(t *testing.T) {
 	})
 }
 
+func waitForListRepositoryRunsLen(ctx context.Context, t *testing.T, repo, ref string, l int) *api.ActionRunList {
+	t.Helper()
+	var runs *api.ActionRunList
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = time.Second * 5
+	bo.MaxElapsedTime = time.Second * 30
+	listFunc := func() error {
+		runsResp, err := client.ListRepositoryRunsWithResponse(ctx, repo, &api.ListRepositoryRunsParams{
+			Commit: api.StringPtr(ref),
+		})
+		require.NoError(t, err)
+		require.Equal(t, http.StatusOK, runsResp.StatusCode())
+		runs = runsResp.JSON200
```

Comment:
Need to check if runs / JSON200 is not nil.  Getting 200 doesn't verify that the content type was json and we managed to parse the result.